fdbclient/BlobCipher.cpp

```diff
@@ -1003,6 +1047,55 @@ StringRef EncryptBlobCipherAes265Ctr::encrypt(const uint8_t* plaintext,
 	return encryptBuf.substr(0, plaintextLen);
 }
 
+void EncryptBlobCipherAes265Ctr::encryptInplace(uint8_t* plaintext,
+                                                const int plaintextLen,
+                                                BlobCipherEncryptHeaderRef* headerRef) {
+	double startTime = 0.0;
+	if (CLIENT_KNOBS->ENABLE_ENCRYPTION_CPU_TIME_LOGGING) {
+		startTime = timer_monotonic();
+	}
+
+	int bytes{ 0 };
+	if (EVP_EncryptUpdate(ctx, plaintext, &bytes, plaintext, plaintextLen) != 1) {
+		TraceEvent(SevWarn, "BlobCipherEncryptUpdateFailed")
+		    .detail("BaseCipherId", textCipherKey->getBaseCipherId())
+		    .detail("EncryptDomainId", textCipherKey->getDomainId());
+		throw encrypt_ops_error();
+	}
+
+	// Padding should be 0 for AES CTR mode, so encryptUpdate() should encrypt all the data
+	if (bytes != plaintextLen) {
+		TraceEvent(SevWarn, "BlobCipherEncryptUnexpectedCipherLen")
+		    .detail("PlaintextLen", plaintextLen)
+		    .detail("EncryptedBufLen", bytes);
+		throw encrypt_ops_error();
+	}
+
+	// we still call EncryptFinal() to be consistent with encrypt() API. It may not be needed.
+	int finalBytes{ 0 };
+	if (EVP_EncryptFinal_ex(ctx, plaintext + bytes, &finalBytes) != 1) {
```

Comment:
It seems risky, as the memory pointed in this method is way past the client allocated buffer?

Reply:
Yeah, that's a good point. As EncryptFinal() doesn't do anything, maybe we could just remove it (and replace it with `EVP_CIPHER_CTX_reset ()`, just in-case the `EncryptBlobCipherAes265Ctr` would be reused).